src/test/kotlin/com/terraformation/backend/tracking/db/ObservationResultsStoreTest.kt

```diff
@@ -410,8 +374,20 @@ class ObservationResultsStoreTest : DatabaseTest(), RunsAsUser {
     return mapCsv(path, func).toMap()
   }
 
-  private fun assertResultsMatchCsv(path: String, func: (Array<String>) -> (() -> Unit)) {
-    val assertions = mapCsv(path, func)
-    assertAll(assertions)
+  private fun assertResultsMatchCsv(
+      path: String,
+      actual: List<List<String>>,
+      mapCsvRow: (List<String>) -> List<String> = { it }
+  ) {
+    val actualRendered = actual.map { it.joinToString(",") }.sorted().joinToString("\n")
+    val expected =
+        mapCsv(path) { mapCsvRow(it.toList()).joinToString(",") }.sorted().joinToString("\n")
+
+    assertEquals(expected, actualRendered, path)
   }
+
+  private fun getSpeciesNameValue(species: ObservationSpeciesResultsPayload): String =
+      species.speciesName ?: species.speciesId?.let { speciesNames[it] } ?: ""
```

Comment:
had to do this in the FE as well... just thinking aloud.. wish kotlin/java had a union construct (speciesName or speciesId)